text/0080-datetime-extension.md

```diff
@@ -0,0 +1,488 @@
+
+* `d`: days
+* `h`: hours
+* `m`: minutes
+* `s`: seconds
+* `ms`: milliseconds
+
+By convention, duration strings should be ordered from largest unit to smallest unit, and contain one quantity per unit. Units with 0 quantity can be omitted. `"1h"`, `"-10h"`, `"5d3ms"`, and `"3m5h"` are all valid duration strings. `"3m5h"`, while valid, is more clearly expressed as `"5h3m"`.
+
+Values of type `duration` have the following methods:
+
+- `.toMilliseconds()` returns a `long` describing the number of milliseconds in this duration. (the value as a long, itself)
+- `.toSeconds()` returns a `long` describing the number of seconds in this duration. (`.toMilliseconds() / 1000`)
+- `.toMinutes()` returns a `long` describing the number of minutes in this duration. (`.toSeconds() / 60`)
+- `.toHours()` returns a `long` describing the number of hours in this duration. (`.toMinutes() / 60`)
+- `.toDays()` returns a `long` describing the number of days in this duration. (`.toHours() / 24`)
+
+Values with type `duration` can also be used with comparison operators:
+
+- `DUR1 < DUR2` returns `true` when `DUR1` is shorter than `DUR2`
+- `DUR1 <= DUR2` returns `true` when `DUR1` is shorter than or equal to `DUR2`
+- `DUR1 > DUR2` returns `true` when `DUR1` is longer than `DUR2`
+- `DUR1 >= DUR2` returns `true` when `DUR1` is longer than or equal to `DUR2`
+- `DUR1 == DUR2` returns `true` when `DUR1` is equal to `DUR2`
+- `DUR1 != DUR2` returns `true` when `DUR1` is not equal to `DUR2`
+
+Equality is based on the underlying representation (see below) so, for example, `duration("1d") == duration("24h")` is true.
+
+#### Representation
+
+The `duration` type is internally represented as a positive or negative quantity of milliseconds as a `long`.
+
+### Operator Overloading
+
+This RFC proposes to use operators `<`, `<=`, `>`, and `>=` for comparing `datetime` and `duration` objects, and `-` for constructing a `duration` from two `datetime`s.
+Currently in Cedar, these operations are only supported for `long`-typed values.
+For other extension types with similar operations, Cedar instead uses extension functions (e.g., `.lessThan()` for decimal values).
+
+This RFC proposes to _reverse_ this decision, and instead allow using builtin operators for extension functions, as appropriate.
+This will add some implementation complexity (at least in the primary [Rust implementation](https://github.com/cedar-policy/cedar)), but it will make policies that use these operations easier to read and easier to write.
+
+### Errors
+
+All the extension functions proposed in this RFC will throw a type error at authorization time if called with the wrong type of arguments.
+Additionally, the `datetime` and `duration` constructors will return an error if the input string does not match the expected format, or if the internal representation of the value (a 64-bit signed int) would overflow. `.offset(duration)` will return an error if the resulting datetime would overflow.
```

Comment:
Probably you should say somewhere that strict validation will require passing literals to `duration` and `datetime` constructors, and it will raise errors if those strings are malformed. Otherwise, validation is straightforward.